packages/renderer/src/lib/statusbar/ProviderWidget.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+import { router } from 'tinro';
+
+import type { ProviderInfo } from '/@api/provider-info';
+
+import IconImage from '../appearance/IconImage.svelte';
+
+let { entry, command = () => router.goto('/preferences/resources') }: { entry: ProviderInfo; command?: () => void } =
+  $props();
+let statusStyle = $state('bg-[var(--pd-status-not-running)] text-[var(--pd-status-not-running)]');
+
+function tooltipText(entry: ProviderInfo): string {
+  let tooltip = '';
+  if (entry.containerConnections.length > 0) {
+    tooltip = entry.containerConnections.map(c => c.name).join(', ');
+  } else if (entry.kubernetesConnections.length > 0) {
+    tooltip = entry.kubernetesConnections.map(c => c.name).join(', ');
+  }
+  return tooltip;
+}
+
+async function executeCommand() {
+  command();
+}
+
+const statusesStyle = new Map<string, string>([
+  ['ready', 'fa-regular fa-circle-check'],
+  ['started', 'fa-regular fa-circle-check'],
+  ['starting', 'animate-spin border border-solid border-[var(--pd-action-button-spinner)] border-t-transparent'],
+  ['stopped', 'fa-regular fa-circle-dot'],
+  ['stopping', 'animate-spin border border-solid border-[var(--pd-action-button-spinner)] border-t-transparent'],
+  ['error', 'fa-regular fa-circle-xmark'],
+  ['not-installed', 'fa fa-exclamation-triangle'],
+  ['update available', 'fa-regular fa-circle-up'],
+  ['unknown', 'fa-regular fa-moon'],
+]);
+
+$effect(() => {
+  if (entry.containerConnections.length > 0) {
+    statusStyle =
+      statusesStyle.get(entry.containerConnections[0].status) ??
+      'bg-[var(--pd-status-not-running)] text-[var(--pd-status-not-running)]';
+  } else if (entry.kubernetesConnections.length > 0) {
+    statusStyle =
+      statusesStyle.get(entry.kubernetesConnections[0].status) ??
+      'bg-[var(--pd-status-not-running)] text-[var(--pd-status-not-running)]';
+  } else {
+    statusStyle = statusesStyle.get(entry.status) ?? 'bg-gray-900 text-[var(--pd-content-text)]';
+  }
+});
```

Comment:
This could be a derived state since only `statusStyle` is modified

Something like the following
```ts
let statusStyle = $devided.by(() => {
  if (entry.containerConnections.length > 0) {
    return
      statusesStyle.get(entry.containerConnections[0].status) ??
      'bg-[var(--pd-status-not-running)] text-[var(--pd-status-not-running)]';
  } else if (entry.kubernetesConnections.length > 0) {
    return
      statusesStyle.get(entry.kubernetesConnections[0].status) ??
      'bg-[var(--pd-status-not-running)] text-[var(--pd-status-not-running)]';
  } else {
    return statusesStyle.get(entry.status) ?? 'bg-gray-900 text-[var(--pd-content-text)]';
  }
})
```